compiler/il/OMRNode.hpp

```diff
@@ -640,10 +645,43 @@ class OMR_EXTENSIBLE Node
 
    TR_YesNoMaybe          hasBeenRun();
 
-   /// Given a monenter node, return the persistent class identifer that's being synchronized
+   /*
+    * The hidden child of a monitor related node is used to store the monitor object's type information.
+    * The type information can be the following two forms:
+    *
+    * 1. The persistent class identifier is stored if the type of a monitor object is a known class.
+    *    Notice java.lang.Object identifier is only stored if it's a fixed class
+    * 2. IDENTITY_CLASS_PLACEHOLDER is stored if the exact monitor type is unknown but is of
+    *    Identity class type
+    */
+   /// Given a monenter node, return the persistent class identifier that's being synchronized
    TR_OpaqueClassBlock *  getMonitorClass(TR_ResolvedMethod *);
+   /*
+    *  \return
+    *    NULL if the concrete class of the monitor object is unknown
+    */
    TR_OpaqueClassBlock *  getMonitorClassInNode();
-   void                   setMonitorClassInNode(TR_OpaqueClassBlock *);
+   /// set the persistent class identifier on the node
+   void                   setMonitorClassInNode(TR_OpaqueClassBlock * classz) { setMonitorInfoInNode(classz); }
+
+   /*
+    * \parm info
+    *    info can be either a persistent class identifier or IDENTITY_CLASS_PLACEHOLDER
+    */
+   void                   setMonitorInfoInNode(void * info);
+   void *                 getMonitorInfoInNode();
+
+   /*
+    * \brief
+    *    whether a monitor object is of identity type
```

Comment:
I think it makes sense to define identity type here in the language agnostic sense of this API - I'm not quite sure what it means so I think I need to understand this to understand how it fits into the OMR design.